lib/semantics/semantics.cpp

```diff
@@ -110,10 +111,11 @@ template<typename... C> class SemanticsVisitor : public virtual C... {
 };
 
 using StatementSemanticsPass1 = ExprChecker;
-using StatementSemanticsPass2 = SemanticsVisitor<AllocateChecker,
-    ArithmeticIfStmtChecker, AssignmentChecker, CoarrayChecker,
-    DeallocateChecker, DoChecker, IfStmtChecker, IoChecker, NullifyChecker,
-    OmpStructureChecker, PurityChecker, ReturnStmtChecker, StopChecker>;
+using StatementSemanticsPass2 =
+    SemanticsVisitor<AllocateChecker, ArithmeticIfStmtChecker,
+        AssignmentChecker, CoarrayChecker, DeallocateChecker, DoChecker,
+        IfStmtChecker, IoChecker, NullifyChecker, OmpStructureChecker,
+        PurityChecker, ReturnStmtChecker, SelectStmtChecker, StopChecker>;
```

Comment:
`SelectConstructChecker` would be a more accurate name, right?  You're not just checking the initial `SELECT` statements.